Kafka, MongoDB, Redis: Publish templated connectors (#324)
mulesoft/docs-connectors: modules/ROOT/pages/redis/redis-connector-studio.adoc

= Redis - Studio - Mule 4
ifndef::env-site,env-github[]
include::_attributes.adoc[]
endif::[]
:imagesdir: ../../assets/images/

== Install the Connector

From within Studio:

. Create a Mule Project.
. In the Mule Palette, click *(X) Search in Exchange*.
. In *Add Modules to Project*, type the name of the connector in the search field.
. Click the connector name in *Available modules*.
. Click *Add >*.
. Click *Finish*.

Or using Exchange:

. In Studio, create a Mule Project.
. Click the Exchange *(X)* icon in the Studio task bar.
. In Exchange, click *Login* and supply your Anypoint Platform username and password.
. In Exchange, select *All assets* and search for "Redis".
. Select Redis, click *Add to project*.
. Follow the prompts to install the connector.

== Configure the Connector

To use Redis connector in your Mule application, configure a global Redis element that can be used by all the connectors in the application.

The sections that follow provide the properties to configure the global element.

== Non-Cluster Configuration

image::redis/redis-basic-config.png[Configuration]

> In the image above, the placeholder values refer to a configuration file placed in the src folder of your project. You can either hardcode your credentials into the global configuration properties, or reference a configuration file that contains these values. For simpler maintenance and better re-usability of your project, Mule recommends that you use a configuration file. Keeping these values in a separate file is useful if you need to deploy to different environments, such as production, development, and QA, where your access credentials differ.

Reviewer: ```suggestion
In the image above, the placeholder values refer to a configuration file placed in the `src` folder of your project. You can either hardcode your credentials into the global configuration properties, or reference a configuration file that contains these values. For simpler maintenance and better re-usability of your project, you should use a configuration file. Keeping these values in a separate file is useful if you need to deploy to different environments, such as production, development, and QA, where your access credentials differ.
```

[%header,cols="25s,75a"]
|===
|Field |Description
|Name | Enter a name for this configuration to reference it later.
|Host| Redis host. Use this field when you want to access a Redis server that is not part of a cluster through the connector.
|Port| Redis port. The port on which your non-cluster server is running.
|Connection Timeout| Connection timeout in milliseconds.

> |Entry Ttl| The default time to live to be set on keys stored through the connector.

Reviewer: The default time to live to be set ?? I don't understand that.

> |Password| Redis password. Set this in case that the server requires authentication.

Reviewer: ```suggestion
|Password| Redis password. Set this in case the server requires authentication.
```

|===

== Cluster Configuration

image::redis/redis-basic-config-clustered.png[Configuration]

[%header,cols="25s,75a"]
|===
|Field |Description
|Name | Enter a name for this configuration to reference it later.
|Cluster Hosts| Redis cluster hosts, separated by commas and with the format host:port. Use this field when you want to access you Redis cluster through the connector.
|Connection Timeout| Connection timeout in milliseconds.

> |Entry Ttl| The default time to live to be set on keys stored through the connector.

Reviewer: The default time to live to be set ?

|===

== Configure an Input Source

Configure an input source for the connector such as a connector operation, using an *HTTP Listener*, or *Scheduler*.

Redis provides the *Subscribe* operation for use as an input source.

`<redis:subscribe>`

[%header,cols="20s,20a,30a,20a,10a"]
|===
| Name | Type | Description | Default Value | Required
| Configuration | String | The name of the configuration to use. | | x
| Channels a| Array of String |  A list of channel names or globbing patterns. |  | x
| Output Mime Type a| String |  The MIME type of the payload that this operation outputs. |  | 

> | Primary Node Only a| Boolean |  Whether this source should only be executed on the primary node when running in  a cluster. |  | 

Reviewer: ```suggestion
| Primary Node Only a| Boolean |  Whether this source should be executed only on the primary node when running in  a cluster. |  | 
```
